### PR TITLE
#1358 Link to 'Settings for etcd tuning' no longer valid, plus other fixes to links to etcd.io/docs

### LINKED ISSUE #1358: Link to "Settings for etcd tuning" no longer valid
## Summary

Within the "[Checklist for Production-Ready Clusters](https://ranchermanager.docs.rancher.com/v2.8/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/checklist-for-production-ready-clusters#networking)" section of the user guide, a link is provided purporting to provide `etcd` tuning recommendations:

etcd-tuning-link

The link points to `https://coreos.com/etcd/docs/latest/tuning.html`, but with coreos.com no longer existing, this now redirects to [etcd.io](https://etcd.io/). Checking archive.org, it looks like this link was invalidated sometime in 2019, as the first instance I can find that has a valid page is back in [2018](https://web.archive.org/web/20181002050041/https://coreos.com/etcd/docs/latest/tuning.html).

I'm not sure the link is still needed given many of the improvements to `etcd` since then, but at the very least, the link should likely be modified to point elsewhere, if not removed.

### PR DESCRIPTION
**Pull request by martyav**
Fixes #1358

## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, remember to add a "MERGE ON RELEASE" label and set the proper milestone.

## Description

From the issue:

> The link points to https://coreos.com/etcd/docs/latest/tuning.html, but with coreos.com no longer existing, this now redirects to [etcd.io](https://etcd.io/). Checking archive.org, it looks like this link was invalidated sometime in 2019, as the first instance I can find that has a valid page is back in [2018](https://web.archive.org/web/20181002050041/https://coreos.com/etcd/docs/latest/tuning.html).

I checked the site the link now redirects to, and found that an updated version of the original Tuning page is now located at https://etcd.io/docs/v3.5/tuning/. I updated the link across the docset, including Chinese language versions.

Since performing the update three other things came up:

- Each link to etcd.io/docs needs to include a version number. There's no default/latest version of the site available without a version number, and at any rate it would be inaccurate if the user needs information on an etcd version that is incompatible with their version of Kubernetes. 
- We had numerous existing links to etcd.io/docs that needed to have their version numbers modified, as [K8s v1.22 and later use etcd v3.5](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.22.md#etcd-moves-to-version-350), and many of our links pointed to pages for earlier etcd versions.
- The etcd.io/docs site used to include trailing zeros in the version number portion of the URL. At some point, the URLs dropped the zeros. In v3.4 and earlier the site would redirect the link correctly if a zero was included, but in v3.5 the outdated links simply throw a `404`.

This PR now addresses both the inaccurate versioning and etcd.io/docs URLs containing trailing zeros across the docset.

### CODE REVIEW

File: i18n/zh/docusaurus-plugin-content-docs/version-2.7/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/checklist-for-production-ready-clusters/checklist-for-production-ready-clusters.md [hunk 1]
```diff
@@ -48,5 +48,5 @@ title: 生产就绪集群检查清单
 
 ### 网络
 
-* 最小化网络延迟。Rancher 建议尽量减少 etcd 节点之间的延迟。`heartbeat-interval` 的默认设置是 `500`，`election-timeout` 的默认设置是 `5000`。这些 [etcd 调优设置](https://coreos.com/etcd/docs/latest/tuning.html) 允许 etcd 在大多数网络（网络延迟特别高的情况下除外）中运行。
+* 最小化网络延迟。Rancher 建议尽量减少 etcd 节点之间的延迟。`heartbeat-interval` 的默认设置是 `500`，`election-timeout` 的默认设置是 `5000`。这些 [etcd 调优设置](https://etcd.io/docs/v3.5/tuning/) 允许 etcd 在大多数网络（网络延迟特别高的情况下除外）中运行。
 * 集群节点应位于单个区域内。大多数云厂商在一个区域内提供多个可用区，这可以提高你集群的可用性。任何角色的节点都可以使用多个可用区。如果你使用 (../set-up-cloud-providers/set-up-cloud-providers.md) 资源，请查阅文档以了解限制（即区域存储限制）。
```
Comment (sunilarjun) [inline i18n/zh/docusaurus-plugin-content-docs/version-2.7/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/checklist-for-production-ready-clusters/checklist-for-production-ready-clusters.md:52]:
```suggestion
* 集群节点应位于单个区域内。大多数云厂商在一个区域内提供多个可用区，这可以提高你集群的可用性。任何角色的节点都可以使用多个可用区。如果你使用 [Kubernetes Cloud Provider](../set-up-cloud-providers/set-up-cloud-providers.md) 资源，请查阅文档以了解限制（即区域存储限制）。
```

File: i18n/zh/docusaurus-plugin-content-docs/version-2.8/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/checklist-for-production-ready-clusters/checklist-for-production-ready-clusters.md [hunk 2]
```diff
@@ -48,5 +48,5 @@ title: 生产就绪集群检查清单
 
 ### 网络
 
-* 最小化网络延迟。Rancher 建议尽量减少 etcd 节点之间的延迟。`heartbeat-interval` 的默认设置是 `500`，`election-timeout` 的默认设置是 `5000`。这些 [etcd 调优设置](https://coreos.com/etcd/docs/latest/tuning.html) 允许 etcd 在大多数网络（网络延迟特别高的情况下除外）中运行。
+* 最小化网络延迟。Rancher 建议尽量减少 etcd 节点之间的延迟。`heartbeat-interval` 的默认设置是 `500`，`election-timeout` 的默认设置是 `5000`。这些 [etcd 调优设置](https://etcd.io/docs/v3.5/tuning/) 允许 etcd 在大多数网络（网络延迟特别高的情况下除外）中运行。
 * 集群节点应位于单个区域内。大多数云厂商在一个区域内提供多个可用区，这可以提高你集群的可用性。任何角色的节点都可以使用多个可用区。如果你使用 (../set-up-cloud-providers/set-up-cloud-providers.md) 资源，请查阅文档以了解限制（即区域存储限制）。
```
Comment (sunilarjun) [inline i18n/zh/docusaurus-plugin-content-docs/version-2.8/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/checklist-for-production-ready-clusters/checklist-for-production-ready-clusters.md:52]:
```suggestion
* 集群节点应位于单个区域内。大多数云厂商在一个区域内提供多个可用区，这可以提高你集群的可用性。任何角色的节点都可以使用多个可用区。如果你使用 [Kubernetes Cloud Provider](../set-up-cloud-providers/set-up-cloud-providers.md) 资源，请查阅文档以了解限制（即区域存储限制）。
```

**Review by sunilarjun — COMMENTED** [review]
Successfully tested the added links, but including possible updates outside of the main scope that could be made to the i18n 2.8/2.7 versions that have a broken link to the "Set Up Cloud Providers" page. Thanks Marty!

**Review by btat — CHANGES_REQUESTED** [review]
1. Older Rancher versions provision clusters with older etcd versions. Since the etcd.io docs are versioned, we should link to the version-appropriate etcd docs instead of updating all our links to point to 3.5.

For example,
- The [v2.6.0 support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-6-0/) lists 
- The [v2.5.0 support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-5-0/) lists etcd 3.3 for downstream clusters while [v2.5.16](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-5-16/) lists etcd 3.4.

2. We have another etcd.io/docs link (on how-to-guides/advanced-user-guides/tune-etcd-for-large-installs) that points to the 3.4 docs. The etcd link version should be consistent within each of our own docs versions.

**Comment by martyav** [comment]
@btat I did a diff of https://etcd.io/docs/v3.3/tuning/ vs https://etcd.io/docs/v3.4/tuning/ and the pages are identical except for the addition of the heading, "When to update the heartbeat interval and election timeout settings", to v3.4. Since in some cases I had to choose between the v3.3 and v3.4 link, I opted for including the earlier version as they're functionally the same. I pushed changes for the English docs. I'll update the international pages and any other pages that include the bad links in a bit

**Comment by martyav** [comment]
@btat I did a crude search on /docs for "https://etcd.io/docs/v3.4" and found 28 instances. Presumably versioned docs will have about as many each. Should we put the updated links for all instances of outdated etcd.io links in this PR or separate them out for ease of review?

**Comment by btat** [comment]
> @btat I did a crude search on /docs for "https://etcd.io/docs/v3.4" and found 28 instances. Presumably versioned docs will have about as many each. Should we put the updated links for all instances of outdated etcd.io links in this PR or separate them out for ease of review?

Given that these should just be version number changes, I think it's fine to include it in this PR since this PR has the context for why it's being done. As for ease of review, if you group things by commit it shouldn't affect readability too much since the actual content change is minimal.

**Review by btat — CHANGES_REQUESTED** [review]
For 2.5 you opted to use the earlier etcd version. For 2.6 the more recent version (3.5) is used, but early 2.6 versions (e.g. [2.6.0](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-6-0/) use 3.4. Is there a content difference between 3.4 and 3.5 to prompt the early vs recent version choice?

**Comment by martyav** [comment]
Just pushed another URL fix, because at about 3.4 the etcd.io site changed how version numbers are written in the URL, from including a trailing zero to not including a trailing zero, and the site would inconsistently redirect links that still contained the zero.

**Review by btat — CHANGES_REQUESTED** [review]
- en: lgtm
- zh: doesn't look like the changes are in sync with the English changes. Some examples:
  - In the `current` set there's a mix of 3.4 and 3.5 links whereas it's all 3.5 on the English side. E.g. `i18n/zh/docusaurus-plugin-content-docs/current/how-to-guides/advanced-user-guides/tune-etcd-for-large-installs.md`
  - On the English side `docs/reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md` is modified, but `i18n/zh/docusaurus-plugin-content-docs/current/reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md` is not.

**Review by btat — APPROVED** [review]

**Pull request merged**